lib/page/redirect-next-page/redirect-next-page.js

```diff
@@ -175,6 +175,29 @@ module.exports = async function getRedirectForNextPage (pageInstance, userData)
       }
     }
 
+    // If user change answer on check answer page and
+    // the user answer goes to an exit page we need to re-valuate the conditions
+    // and redirect the user to the exit page.
+    //
+    // Caveat: Save and return feature is considered an exit page. We don't
+    // want to halt execution when saving your progress. Then we continue the
+    // flow when the next page contains save and return scope.
+    //
+    if (checkNextPage && checkNextPage._type === 'page.exit' && checkNextPage.scope !== 'savereturn') {
```

Comment:
why `scope` and not`_type`?

Reply:
Because save and return pages are page.exit types hence the bug. I have to use scope as it is the only indicative of that is a save and return.